Source/NSLayoutConstraint.m

```diff
@@ -0,0 +1,578 @@
+/* Implementation of class NSLayoutConstraint
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory Casamento <greg.casamento@gmail.com>
+   Date: Sat May  9 16:30:22 EDT 2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSKeyedArchiver.h>
+
+#import "AppKit/NSControl.h"
+#import "AppKit/NSView.h"
+#import "AppKit/NSAnimation.h"
+#import "AppKit/NSLayoutConstraint.h"
+#import "AppKit/NSWindow.h"
+#import "AppKit/NSApplication.h"
+
+static NSMutableArray *activeConstraints = nil;
+static NSNotificationCenter *nc = nil;
+
+@implementation NSLayoutConstraint
+
++ (void) initialize
+{
+  if (self == [NSLayoutConstraint class])
+    {
+      [self setVersion: 1];
+      activeConstraints = [[NSMutableArray alloc] initWithCapacity: 10];
+      nc = [NSNotificationCenter defaultCenter];
+
+      [nc addObserver: self
+             selector: @selector(_setupNotifications:)
+                 name: NSApplicationDidFinishLaunchingNotification
+               object: nil];
+    }
+}
+
++ (void) _setupNotifications: (NSNotification *)n
+{
+  NSEnumerator *en = [[NSApp windows] objectEnumerator];
```

Comment:
That way constraints will only work on the initial windows. Looks like we need a better way to connect a window to the  corresponding constraints and only try to solve the relevant constraints for each window.

Reply:
Ugh, you are correct.  I want to keep all code related to constraints in this file only (I don't want to make changes to NSWindow and such to support this).  I will find another way.

Reply:
I may have to add a method to "evaluate" the constraints prior to anything else.  The reason for this is that, in some cases, the constraints can disable the ability of the window to resize.  I need to determine this situation when the application comes up.

Reply:
I believe this is discussed below.  I need to make sure everything is being decoded correctly.  I will resolve this for now as it is covered by the later conversation.